frmts/avif/avifdataset.cpp

```diff
@@ -342,6 +362,96 @@ GDALAVIFIO::GDALAVIFIO(VSIVirtualHandleUniquePtr fpIn) : fp(std::move(fpIn))
     return AVIF_RESULT_OK;
 }
 
+#ifdef AVIF_HAS_OPAQUE_PROPERTIES
+/************************************************************************/
+/*                          GetSpatialRef()                             */
+/************************************************************************/
+const OGRSpatialReference *GDALAVIFDataset::GetSpatialRef() const
+{
+    return geoHEIF.GetSpatialRef();
+}
+
+/************************************************************************/
+/*                          GetGeoTransform()                           */
+/************************************************************************/
+CPLErr GDALAVIFDataset::GetGeoTransform(double *padfTransform)
+{
+    return geoHEIF.GetGeoTransform(padfTransform);
+}
+
+void GDALAVIFDataset::processProperties()
+{
+    for (size_t i = 0; i < m_decoder->image->numProperties; i++)
+    {
+        avifImageItemProperty *prop = &(m_decoder->image->properties[i]);
+        if (!memcmp(prop->boxtype, "mcrs", 4))
+        {
+            geoHEIF.extractSRS(prop->boxPayload.data, prop->boxPayload.size);
+        }
+        else if (!memcmp(prop->boxtype, "mtxf", 4))
+        {
+            geoHEIF.setModelTransformation(prop->boxPayload.data,
+                                           prop->boxPayload.size);
+        }
+        else if (!memcmp(prop->boxtype, "tiep", 4))
+        {
+            geoHEIF.addGCPs(prop->boxPayload.data, prop->boxPayload.size);
+        }
+        else if (!memcmp(prop->boxtype, "udes", 4))
+        {
+            extractUserDescription(prop->boxPayload.data,
+                                   prop->boxPayload.size);
+        }
+    }
+}
+
+void GDALAVIFDataset::extractUserDescription(const uint8_t *payload,
+                                             size_t length)
+{
+    // Match version
+    if (payload[0] == 0x00)
```

Comment:
perhaps add a CPLDebug() message in the else clause about version mismatch (applies to other similar situations)

Reply:
Added in relevant places.